capstone/capapi/templates/registration/login.html

```diff
@@ -25,7 +25,7 @@
             class="btn btn-primary">
       Log in
     </button>
-    <input type="hidden" name="next" value="{{ next }}" />
+    <input type="hidden" name="next" value="{{ request.META.HTTP_REFERER }}" />
```

Comment:
This won't work if the user enters the wrong password the first time, or if the login form is linked from off-site. Maybe a better fix for the open issue would be to add `?next={{ request.get_full_path }}` to the login link in the header?

Reply:
ACK, great catch!!